### DISCUSSION TITLE
Source sub project "pinger"

### DISCUSSION THREAD

**Discussion by JamesOlvertone** [zlatinb/muwire#166] [Q&A]
What is this for? Pinging other users? Why is the second command line option a file?
The base64 address is the long user id?

**Comment by zlatinb** [comment]
It's a tool that sends the contents of a file to the specified b64 destination (not long user id).  I used it while developing the "host-cache" component.  It has had virtually no use since.

**Comment by JamesOlvertone** [comment]
Yes I discovered that later by myself. :) Started the host-cache programm and pinged it and it responded on stdout, the file that I sent was in wrong format but that doesn't matter, now I know what it is for.

Is this host cache programm useful in any way if the address of my host-cache is not included in the official source?
After start up it tries to crawl for nodes, but never found any (running some hours).

**Reply by zlatinb** [reply]
The host-cache is intended to be included in the source.  The host cache learns about nodes from the pings it receives and then it starts crawling.  If it has never been pinged by a muwire node that is connected to the live network the host cache will not discover any nodes.

See the [wiki page](https://github.com/zlatinb/muwire/wiki/MuWire-HostCache) as well as the `core/src/main/groovy/com/muwire/core/hostcache/CacheServers.groovy` source file.
